File: server/features/terminal.go
package features

import "github.com/tstapler/stapler-squad/server/featureregistry"

// TerminalRender describes the terminal streaming RPC and its UI surface.
var TerminalRender = featureregistry.Feature{
	ID:          "terminal-render",
	Title:       "Terminal Rendering",
	Description: "Streams terminal output to the browser with RAF batching.",
	RPCIDs:      []string{"session:stream-terminal"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

func init() {
	featureregistry.Register(TerminalRender)
}

File: web-app/src/lib/features/features/terminal.ts
import type { Feature } from '../types';

export const terminalFeatures = {
  'terminal-render': {
    id: 'terminal-render',
    title: 'Terminal Rendering',
    description: 'Streams terminal output to the browser with RAF batching.',
    rpcIds: ['session:stream-terminal'],
    componentPaths: ['components/terminal/Terminal.tsx'],
    testIds: [
      'Terminal Flickering Fix > should maintain 60fps with incremental rendering',
    ],
    status: 'stable',
    since: '1.0.0',
  },
} as const satisfies Record<string, Feature>;
